### ISSUE TITLE
Checkout Block: Small delay when updating Shipping Methods

### ISSUE THREAD

**Issue by jimjasson** [woothemes/woocommerce#42235]
## Describe the bug

When changing the value of a checkout field that triggers the update of the Shipping Methods, then there are a couple of seconds before this update when the Place Order button remains clickable. If someone clicks this button before the Shipping Methods had a chance to get updated, the order is not placed. 

https://github.com/woocommerce/woocommerce-blocks/assets/25955483/9a81d5cc-b8a3-41ef-a587-899c2f2262b8

https://github.com/woocommerce/woocommerce-blocks/assets/25955483/f0afe7aa-b351-46c6-bc44-f9073b900b7c

 
## To reproduce

1. Add a product to the cart and navigate to the block-based checkout.
2. Fill in the form.
3. Scroll down so that both the zip code field and the Place Order button are in the viewport.
4. Change the last digit of the zip code and immediately click the Place Order button.

## Expected behavior

The Place Order button should be disabled or the update of the Shipping Methods should happen quicker.

### WordPress (please complete the following information):

-   WordPress version: 6.3
-   WooCommerce version: 8.0.2
-   WooCommerce Blocks version: 10.9.3

**Comment by tarunvijwani** [comment]
@jimjasson I'm unable to reproduce the bug. The `place order` button gets disabled for me when I use the storefront theme, just like in your first video. Also, I couldn't replicate the issue where the button doesn't get disabled with any other theme. 



https://github.com/woocommerce/woocommerce-blocks/assets/11503784/510af208-b804-4f39-b6af-81860f209f70


Could you please let me know the name of the theme you used in the second video and confirm again whether you were able to place an order while the shipping methods were loading? 
Any additional information you can provide would be very helpful.

**Comment by jimjasson** [comment]
Hey @tarunvijwani , 

Thank you for checking! I was using the Tsubaki theme in the original videos, but I can replicate this issue with Storefront as well: 


https://github.com/woocommerce/woocommerce-blocks/assets/25955483/b9aff068-fd2d-49a6-bd68-d8bdb64f7d9f

Try this: 
- Click to update the zip code. 
- Move your cursor above the Place Order button. 
- Using your keyboard, delete the last digit of the zip code and enter a new one. 
- Click the Place Order button immediately.

If you still cannot reproduce this issue, try adding some network delay in your browser's settings.

**Comment by tarunvijwani** [comment]
@jimjasson Thank you for sharing the screen recording. I followed your instructions and activated the Tsubaki theme and added the network delay, but I wasn't able to replicate the issue you reported.

https://github.com/woocommerce/woocommerce-blocks/assets/11503784/e8317c7d-376f-4ff4-82c2-86a27eab34cf

From what I understand from your video that if a user directly clicks on the "place order" button while editing the fields, it should not place an order.  I see that we already have mechnasim in place to prevent the checkout and you won't be able to place order while checkout is still loading or about to load the shipping method. Please correct me if you're able to successfully place an order.

**Comment by tarunvijwani** [comment]
I don't think this is a bug. Let's loop in @senadir, @opr, and @mikejolley for a second opinion. 🙂

**Comment by jimjasson** [comment]
Hey @tarunvijwani, 

The issue is not that an order is placed when shipping methods are still loading. The issue is with the UX; there is a button that becomes "active" but if users click on it, it does nothing. This button should be disabled while the form is still updating. 

As you can see in the second video in the issue description when I added a network delay, the Place Order button remained visible when the shipping options were updated. However, when I do the same experiment now, I see that the Place Order button gets disabled when the shipping options are updated, which is the expected behavior. 

Is there a chance that you changed something in the logic that determines when this button gets disabled?

**Comment by opr** [comment]
I see the `disabled` attribute on the button when changing the address, I don't see a visual indicator that it's disabled though, maybe we should add that.

> Is there a chance that you changed something in the logic that determines when this button gets disabled?

I checked all places where we `__internalIncrementCalculcating`. The `calculatingCount` is increased whenever a server-side operation is taking place, and it is decreased when the server responds. This is in turn used in the Checkout data store to decide whether `isCalculating` is true. `isCalculating` is used by `useCheckoutSubmit` to determine if the button should be disabled.

I couldn't see any edits to this flow since 9th August https://github.com/woocommerce/woocommerce-blocks/pull/10315 but this was opened on 29th August so I'm sure you were seeing those changes when you reported this.

Not sure what else could have changed in the meantime though!

**Comment by opr** [comment]
Hi @jimjasson I tested this again, the place order button becomes immediately disabled for me when changing anything in the address.

I changed my city and _immediately_ pressed place order before `pushChanges` was triggered and the checkout process did not start (as expected). After the update I could check out with the correct address.

I'll close this as I don't think it's valid anymore.
